airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/api_adapter.py

```diff
@@ -0,0 +1,275 @@
+            ("placements", "placement"),
+            ("inventory_filter_enabled", "enable_inventory_filter"),
+            ("promotion_type", "external_type"),
+            ("optimization_event", "external_action"),
+            ("secondary_optimization_event", "deep_external_action"),
+            ("age_groups", "age"),
+            ("operating_systems", "operation_system"),
+            ("network_types", "connection_type"),
+            ("device_price_ranges", "device_price"),
+            ("min_android_version", "android_osv"),
+            ("ios14_targeting", "ios_target_device"),
+            ("min_ios_version", "ios_osv"),
+            ("optimization_goal", "optimize_goal"),
+            ("bid_price", "bid"),
+            ("conversion_bid_price", "conversion_bid"),
+            ("deep_cpa_bid", "deep_cpabid"),
+            ("next_day_retention", "daily_retention_ratio"),
+            ("secondary_status", "status"),
+            ("operation_status", "opt_status"),
+            ("actions", "action_v2"),
+            ("video_user_actions", "user_actions"),
+            ("rf_purchased_type", "rf_buy_type"),
+            ("purchased_impression", "buy_impression"),
+            ("purchased_reach", "buy_reach"),
+            ("rf_estimated_cpr", "rf_predict_cpr"),
+            ("rf_estimated_frequency", "rf_predict_frequency"),
+            ("included_custom_actions", "include_custom_actions"),
+            ("excluded_custom_actions", "exclude_custom_actions"),
+            ("shopping_ads_retargeting_type", "dpa_retargeting_type"),
+            ("brand_safety_type", "brand_safety"),
+            ("expansion_enabled", "enable_expansion"),
+            ("adgroup_app_profile_page_state", "ad_app_profile_page_type"),
+            ("delivery_mode", "display_mode"),
+            ("comment_disabled", "is_comment_disable"),
+            ("store_authorized_bc_id", "store_authorized_bc"),
+            ("audience_ids", "audience"),
+            ("excluded_audience_ids", "excluded_audience"),
+            ("location_ids", "location"),
+            ("interest_category_ids", "interest_category_v2"),
+            ("interest_keyword_ids", "interest_keywords"),
+            ("device_model_ids", "device_models"),
+            ("carriers_ids", "carriers_v2"),
+            ("video_download_disabled", "video_download"),
+            ("blocked_pangle_app_ids", "pangle_block_app_list_id"),
+            ("action_category_ids", "action_categories"),
+            ("included_pangle_audience_package_ids", "pangle_audience_package_include"),
+            ("excluded_pangle_audience_package_ids", "pangle_audience_package_exclude"),
+            ("catalog_authorized_bc_id", "catalog_authorized_bc"),
+            ("auto_targeting_enabled", "automated_targeting"),
+        ),
+        "report/integrated/get/": (("metrics/placement_type", "metrics/placement"),),
+    }
+    _type_changed_fields_by_stream = {
+        "ads": (
+            ("advertiser_id", int),
```

Comment:
is this change safe? do we know these can always be adjusted to int?

Reply:
No guarantees. TikTok says they changed the data type of IDs for the sake of compatibility, so I believe this is relatively safe

![image](https://user-images.githubusercontent.com/7703835/185734665-5f78fc9d-251a-4a83-89c4-0d8a0e08c598.png)

our options here are:
 - accept this change up until the incident :shrug: 
 - stay safe: do not adjust the data type but change the schema
 - go further - apply all the API changes to schema (then I'd close this PR and open a new one)